zerver/tests/test_signup.py

```diff
@@ -1610,6 +1609,41 @@ def test_accessing_invites_in_another_realm(self) -> None:
         error_result = self.client_delete('/json/invites/' + str(prereg_user.id))
         self.assert_json_error(error_result, "No such invitation")
 
+    def test_prereg_user_status(self) -> None:
+        email = self.nonreg_email("alice")
+        password = "password"
+        realm = get_realm('zulip')
+        realm.email_address_visibility = Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE
+        realm.save(update_fields=['email_address_visibility'])
+
+        inviter = UserProfile.objects.first()
+        prereg_user = PreregistrationUser.objects.create(
+            email=email, referred_by=inviter, realm=realm)
+
+        confirmation_link = create_confirmation_link(prereg_user, 'host', Confirmation.USER_REGISTRATION)
+        registration_key = confirmation_link.split('/')[-1]
+
+        confirmation = Confirmation.objects.get(confirmation_key=registration_key)
+        prereg_user = confirmation.content_object
+        active_value = confirmation_settings.STATUS_ACTIVE
+
+        result = self.client_post(
+            "/accounts/register/",
+            {"key": registration_key,
+             "from_confirmation": "1",
+             "full_name": "alice"})
+        self.assertEqual(prereg_user.status, 0)
+        self.assertEqual(result.status_code, 200)
+
+        result = self.submit_reg_form_for_user(email, password, key=registration_key)
+        self.assertEqual(result.status_code, 302)
+        prereg_user = PreregistrationUser.objects.get(
+            email=email, referred_by=inviter, realm=realm)
+        self.assertEqual(prereg_user.status, active_value)
```

Comment:
Let's just do `confirmation_settings.STATUS_ACTIVE` here, we dont need the extra variable